### DISCUSSION TITLE
temp files using up all disk space

### DISCUSSION THREAD

**Discussion by deltaforce** [ImageMagick/ImageMagick#3620] [Help]
On one of my sites, I am constantly getting disk space warnings because the home/cagefs/tmp folder is being filled up with imagic files.
I do not know how or why imagemagick is even being used, as nobody is doing anything on this website, no posts, no editing, no uploading images, literally nothing at all has been done or changed on this site for over 1 year.

Why are these imagic files being created? how can I find out what is causing it?

**Comment by snibgo** [comment]
I suppose you mean files with names like "magick-64365HCLJ39mDwsr".

IM creates such files internally, and deletes them. But if IM is killed externally, for example because externally-imposed resource limits are exceeded, it has no chance to delete those files.

If you are certain that IM should not be used, then you could remove it and see what breaks.

But perhaps IM is not installed, but is embedded in some other software.

I suggest you delete these files, perhaps daily.

**Comment by deltaforce** [comment]
I am of course deleting these files whenever I get a notification.
But as I said, nobody is doing anything on this website, no posts, no editing, no uploading images, literally nothing at all has been done or changed on this site for over 1 year.
So I have no idea how or why imagemagick is being used or executed to begin with.
There is no person or software doing any image manipulation nor are there any images that need to be manipulated  any way, because nothing has changed on the site in a very long time. The last time a new image was uploaded to the media library was over 2 years ago.

**Reply by fmw42** [reply]
Perhaps there is a cron job running that uses ImageMagick.

**Reply by deltaforce** [reply]
nope, the only cron job I have is one which sends out emails to subscribers

**Reply by fmw42** [reply]
Does it include graphics? Perhaps it is calling ImageMagick to process some graphic?

**Reply by fmw42** [reply]
What is an example of a file name that you are getting?  Does it start with magick or imagick or something else.

**Reply by deltaforce** [reply]
magick-****

**Reply by fmw42** [reply]
Some other tool is then using ImageMagick.

**Reply by deltaforce** [reply]
And the sky is blue.... but pointing out the obvious doesn;t help.... I posted this as I want to resolve the issue, not just have the obvious pointed out.
 the only thing installed on this account is 1 x wordpress site, nothing else.

**Reply by fmw42** [reply]
Wordpress uses ImageMagick

**Reply by fmw42** [reply]
See for example, https://wordpress.org/plugins/imagemagick-engine/

**Reply by deltaforce** [reply]
I obviously  know wordpress uses Imagemagick, otherwise I wouldn't be here posting this question.

Imagemagick is a php extension which is enabled via the php.ini file or via cpanel.
When you enable it, it then becomes available for WordPRess for any site to use.

**Comment by snibgo** [comment]
Is ImageMagick installed? If so, what happens if you remove it?

**Reply by deltaforce** [reply]
I cannot remove it, I can only disable it in cpanel/php options. Which I have now done

**Reply by snibgo** [reply]
According to the WordPress page that fmw42 linked, WordPress will use the "standard GD image library" if ImageMagick is not available. So I suppose that disabling IM won't break anything.

**Reply by deltaforce** [reply]
yes that is correct, GD is the default image library that WP will use, but it is slow, which is why people tend to use Imagemagick.
After I had this issue I went and check a while bunch of other sites and they all had these files consuming space, just not enough to result on using up all the disk space quota
